src/main/java/com/icthh/xm/ms/entity/repository/XmEntityRepository.java

```diff
@@ -30,130 +30,162 @@
         return springXmEntityRepository.existsByTypeKeyAndNameIgnoreCase(typeKey, name);
     }
 
+    @Override
     public List<XmEntity> findAll() {
         return springXmEntityRepository.findAll();
     }
 
+    @Override
     public List<XmEntity> findAll(Sort sort) {
         return springXmEntityRepository.findAll(sort);
     }
 
-    public List<XmEntity> findAll(Iterable<Long> longs) {
-        return springXmEntityRepository.findAll(longs);
+    @Override
+    public List<XmEntity> findAllById(Iterable<Long> longs) {
+        return springXmEntityRepository.findAllById(longs);
     }
 
+    @Override
     public void flush() {
         springXmEntityRepository.flush();
     }
 
+    @Override
     public void deleteInBatch(Iterable<XmEntity> entities) {
         springXmEntityRepository.deleteInBatch(entities);
     }
 
+    @Override
     public void deleteAllInBatch() {
         springXmEntityRepository.deleteAllInBatch();
     }
 
+    @Override
     public XmEntity getOne(Long aLong) {
         return springXmEntityRepository.getOne(aLong);
     }
 
+    @Override
     public <S extends XmEntity> List<S> findAll(Example<S> example) {
         return springXmEntityRepository.findAll(example);
     }
 
+    @Override
     public <S extends XmEntity> List<S> findAll(Example<S> example, Sort sort) {
         return springXmEntityRepository.findAll(example, sort);
     }
 
+    @Override
     public Page<XmEntity> findAll(Pageable pageable) {
         return springXmEntityRepository.findAll(pageable);
     }
 
-    public XmEntity findOne(Long aLong) {
-        return springXmEntityRepository.findOne(aLong);
+    @Override
+    public Optional<XmEntity> findById(Long aLong) {
+        return springXmEntityRepository.findById(aLong);
     }
 
-    public boolean exists(Long aLong) {
-        return springXmEntityRepository.exists(aLong);
+    @Override
+    public boolean existsById(Long aLong) {
+        return springXmEntityRepository.existsById(aLong);
     }
 
+    @Override
     public long count() {
         return springXmEntityRepository.count();
     }
 
-    public void delete(Long aLong) {
-        springXmEntityRepository.delete(aLong);
+    @Override
+    public void deleteById(Long aLong) {
+        springXmEntityRepository.deleteById(aLong);
     }
 
+    @Override
     public void delete(XmEntity entity) {
         springXmEntityRepository.delete(entity);
     }
 
-    public void delete(Iterable<? extends XmEntity> entities) {
-        springXmEntityRepository.delete(entities);
+    @Override
+    public void deleteAll(Iterable<? extends XmEntity> entities) {
+        springXmEntityRepository.deleteAll(entities);
     }
 
+    @Override
     public void deleteAll() {
         springXmEntityRepository.deleteAll();
     }
 
-    public <S extends XmEntity> S findOne(Example<S> example) {
+    @Override
+    public <S extends XmEntity> Optional<S> findOne(Example<S> example) {
```

Comment:
ALARM! It's can broke leps!
Please resolve optional inside repository

Reply:
`<S extends XmEntity> Optional<S> findOne(Example<S> example)` clashes with `public <S extends XmEntity> S findOne(Example<S> example)`

but it seems this method is not used inside LEPs (in meanfind by Example, all the cases are using findByID)

Reply:
some body could use it, and it can broke old tenant

Reply:
done